### ISSUE TITLE
website content to be translated

### ISSUE THREAD

**Issue by maddhruv** [nodejs/i18n#72]
As also discussed in the previous meeting
How shall we be translating and synchronizing the website contents?

It will be beneficial proposing a robust way of doing that with @nodejs/website-redesign and wrt https://github.com/nodejs/website-redesign/issues/31

A centralized content repo for having contents for the website and i18n access and if @nodejs/website-redesign chooses a CMS for the same.

**Comment by lukaszewczak** [comment]
Hi @maddhruv , could please describe in more details your idea of the centralized content repo?

**Comment by lukaszewczak** [comment]
Hi, If I understand correctly, then the method of translating the content for a website depends very much on how the new web page will be built, but in the case of api documentation, nothing changes? Perhaps let's start with a focus on how to translate api and how to solve the issue with translating different api node.js versions.

**Comment by maddhruv** [comment]
Hey, sorry for the late reply, was busy with other stuff.

The idea behind a centralized repo is to maintain all the contents of the _new website_, plus the docs and other wiki pages, so as it would be easy enough to serve them to the website and also for crowdin integration, we would only have to give access to this central repo. Also it would be much easier to translate the website content and docs.

**Comment by fhemberger** [comment]
As the redesign process is still in its early stages, it would be great if you could help us in the mean time setting up Crowdin for the current @nodejs/website. There is a lot of translating already happening right now, and we sure could need a better process to handle updates. ;)

**Comment by zeke** [comment]
@fhemberger I'd be interested in exploring this. Does the website group meet regularly? If so, I can make an effort to join the next meeting to discuss this.

**Comment by fhemberger** [comment]
@zeke No, we don't have any sort of team meetings (we have > 160 contributors 😁). Everything is discussed in GitHub issues.

**Comment by fhemberger** [comment]
@nodejs/i18n *bump*

**Comment by Andrulko** [comment]
@fhemberger If your translators team have any questions, I will be happy to answer / join a meeting. Not sure if you've tried our In-Context for website translation:
https://support.crowdin.com/in-context-localization/

**Comment by outsideris** [comment]
Now, we are using the crowdin for translating our website?

**Comment by fhemberger** [comment]
@outsideris No, not yet. But translation became a huge part of the website's PRs, so I think we could profit from joining whatever is used for translating the docs. Also to have a better comparison which translations are not up-to-date, etc.

**Comment by fhemberger** [comment]
@zeke Awesome, looking forward to it!

**Comment by fhemberger** [comment]
@zeke BTW, are changes on Crowdin synced back to the Node.js docs in the main repo?
If so: Could we add the current website as a second project? Just to try out the workflow on some live content?

Translations are all in https://github.com/nodejs/nodejs.org/tree/master/locale

**Comment by zeke** [comment]
> are changes on Crowdin synced back to the Node.js docs in the main repo?

The source English content comes from https://github.com/nodejs/node/tree/master/doc, and is fetched for all currently supported versions of Node.js

Crowdin opens PRs on this repo (node.js/i18n) that add translated Markdown files in other languages. The goal is for the new website to consume the contents of this repo as a node module. See https://github.com/nodejs/i18n/issues/92 - Here's a WIP PR to build such a module: https://github.com/nodejs/i18n/pull/133

**Comment by glaucia86** [comment]
@maddhruv @fhemberger I do not know if you're noticing, but the Reactjs.org team tried to do the translations through Crowdin. But without success.
So what they decided to do? They opened the following issue https://github.com/reactjs/reactjs.org/issues/1605
where they organized by language and called on the community of the whole world to help with translations.
Those responsible for approving and validating translations are the members and contributors of the Organization. What is good to ensure good translation.

Since then, there are several people from all over the world contributing and helping in these translations.

I think if we just make and leave those translations to do at Crowdin, we'll take a long time to finish.

To give you an idea, the translation of en the React, has started two weeks and we are almost finalizing all of it: https://github.com/reactjs/pt-BR.reactjs.org/issues/1

I think if we engage the entire technical community, following the excellent example that the Reactjs.org and Vuejs.org team did, we will have the Nodejs.org website fully translated into several languages.

I am in favor of following their example. What do you think?

cc: @cezaraugusto @zeke  @bnb @MylesBorins 


![screen shot 02-14-19 at 10 25 am](https://user-images.githubusercontent.com/1631477/52871640-a735f300-3131-11e9-9aee-661616274b45.PNG)

**Comment by zeke** [comment]
> I think if we just make and leave those translations to do at Crowdin, we'll take a long time to finish.

I disagree. The Electron project has over 1000 volunteer translators, and we're seeing great progress in numerous languages: https://electronjs.org/languages

From https://github.com/reactjs/reactjs.org/issues/1605

> each translation is managed in a separate fork of the main repo by a dedicated team

This is existing unscalable approach used by the @nodejs org. One of the benefits of using Crowdin is that all content in all languages is guaranteed to be up to date, even if not fully translated. When using GitHub alone with forks and branches, content gets out of date very quickly.

We've already gone to great lengths to set up this nodejs/i18n translation pipeline. The remaining work is to make the translations in this repo easily consumable by other projects, namely the Node.js website.

**Comment by glaucia86** [comment]
> I disagree. The Electron project has over 1000 volunteer translators, and we're seeing great progress in numerous languages: https://electronjs.org/languages
> 
> From [reactjs/reactjs.org#1605](https://github.com/reactjs/reactjs.org/issues/1605)
> 
> > each translation is managed in a separate fork of the main repo by a dedicated team
> 
> This is existing unscalable approach used by the @nodejs org. One of the benefits of using Crowdin is that all content in all languages is guaranteed to be up to date, even if not fully translated. When using GitHub alone with forks and branches, content gets out of date very quickly.
> 
> We've already gone to great lengths to set up this nodejs/i18n translation pipeline. The remaining work is to make the translations in this repo easily consumable by other projects, namely the Node.js website.

ok. I just give a other option. What all of you guys deciced it. I will acept it. Thanks.

cc: @zeke

**Comment by alexandrtovmach** [comment]
@zeke Hello, really appreciate your help with Crowdin project configuration. Can I help or assist you with it to finally make possible to translate website content too?

**Comment by alexandrtovmach** [comment]
I had a chat with CrowdIn team, and we discussed the solution for nodejs website. Here is conclusion:

1. Need to create and configure one more project on CrowdIn by _nodejs-crowdin_ user, for example ― *nodejs-website* which will be based on [https://github.com/nodejs/nodejs.org](https://github.com/nodejs/nodejs.org)
2. Probably rename current [Nodejs](https://crowdin.com/project/nodejs) project to *nodejs-docs*

**Comment by alexandrtovmach** [comment]
*Summarizing:*

Currently, all translations of the website content @nodejs/website  goes through PR and all API docs @nodejs/documentation CrowdIn. From my point of view the best option is to move translation process of the website to CrowdIn too, to have one centralized place with all translation data. Later, when @nodejs/website-redesign team complete new version, we can easily migrate to new repository, without breaking translation process.

*To do*
I'm ready to investigate, but I need to get permissions/credentials on CrowdIn to configure system for website translation. @zeke could you help me with that?

**Comment by zeke** [comment]
@alexandrtovmach yeah I can help with that. I don't currently have much free time to work on this, but I can help unblock. Someone needs to:

- Update the outdated list of target node versions in: https://github.com/nodejs/i18n/blob/6a0903d733f7b2bbaae0159e5c0bf564b90b7b8c/package.json#L5-L8
- Run this script: https://github.com/nodejs/i18n/blob/master/script/collect.js
- Open a PR with the changes

That will get the latest docs into the repo, and they should automatically integrated into the existing Crowdin pipeline for translation.

After that, the question is how to get the new website to consume the translated data from the nodejs/i18n repo.

**Comment by alexandrtovmach** [comment]
@zeke So I've collected the newest content and here's PR. But what about integrating nodejs.org content with CrowdIn as I described.

**Comment by zeke** [comment]
Once the changes from your PR https://github.com/nodejs/i18n/pull/199 land on master, they'll automatically be ingested into Crowdin, and Crowdin will soon after create a followup PR that updates the translated files to be in sync with the source English.

**Comment by alexandrtovmach** [comment]
@zeke, I see but that's only docs l, but I asked about content for a website.

**Comment by zeke** [comment]
Oh I'm sorry I misread your comment. There are a few ways we could do this:

### Option 1: Create a second Crowdin project

Like you mentioned in https://github.com/nodejs/i18n/issues/72#issuecomment-507218070, we could create a second crowdin project. This would be relatively easy to set up, but could introduce issues with managing access/membership/proofreaders/managers/isuses/etc. I don't think Crowdin has a concept of "orgs" which would allow shared teams or permissions across multiple Crowdin projects. @Andrulko might have some insight for us on this.

### Option 2: Collect website content into nodejs/i18n

Write a script similar to the existing `script/collect.js` which pulls in content from the website repo. This would allow the existing Crowdin integration with node.js/i18n to continue functioning. We'd still need to figure out a way for the website to consume the translated content from this repo.

### Option 3: Move everything to the website repo

We could change the Crowdin integration to sync directly with the website repo, and phase out this nodejs/i18n repo, moving all of the API docs and docs collection scripts directly into the website repo.

All of these options come with their own set of tradeoffs. I have not had much time to work on this lately, so I will defer to others (especially those working on the website) to help decide the best path forward.

**Comment by alexandrtovmach** [comment]
@zeke Here is a lot of options, but from my point of view better to go with *Option #1*. CrowdIn give us a possibility to manage roles less than Manager _per project_ not per organization, and I think it's okay. Currently, we have the same structure, where one part of collaborators working on website another on documentation, so here're no issues, I suppose.

CrowdIn haven't instances like organization, but have *Owner* and *Manager*. We already have owner: _nodejs-crowdin_ and need to add managers. Manager can be assigned to only one project or to a few:
- [documentation](https://support.crowdin.com/assigning-managers-to-several-projects/)
- [example](https://crowdin.com/profile/mxvideoplayer)

**Comment by alexandrtovmach** [comment]
My account on CrowdIn is `alexandrtovmach` so if we decided to go with _Option 1_ you can create a new project for a website and give me a Manager rights to I can set up translation process with [nodejs.org repo](https://github.com/nodejs/nodejs.org).

**Comment by zeke** [comment]
That sounds okay to me, @alexandrtovmach. I've reached out to @Andrulko on our private Slack channel for insight on how to best move forward.

**Comment by alexandrtovmach** [comment]
@zeke Okay, cool

**Comment by Andrulko** [comment]
@zeke @alexandrtovmach 

In your scenario, since currently content is hosted in a separate repositories on GitHub + you would like to automate sync with the help of Crowdin<->GitHub integration, the most optimal way would be another project connected with https://github.com/nodejs/nodejs.org repository.

Translation Memory will be shared across projects and using [Advanced Workflows](https://support.crowdin.com/advanced-workflows/) you can apply 100% TM matches automatically (**Important!** Workflow should be configured under the main account in Crowdin, **nodejs-crowdin**).

Probably not everyone who is involved in docs translation will be ready to join website translation, so volunteers can always decide in which project they wish to contribute :)

Please drop me a line if any additional questions arise!

**Comment by zeke** [comment]
Thanks for the insight, @Andrulko.

@alexandrtovmach I'll be happy to set you up on Crowdin. Can you first open a PR here to add yourself to the nodejs/i18n working group? https://github.com/nodejs/i18n#joining

**Comment by alexandrtovmach** [comment]
@Andrulko Great, thanks for the explanation, appreciate it

@zeke Done #202

**Comment by zeke** [comment]
@alexandrtovmach thanks for joining the group, and for taking the initiative here. I've invited you as a manager of the https://crowdin.com/project/nodejs project.

**Comment by alexandrtovmach** [comment]
@zeke Amazing, thank you. I'm start working on it

**Comment by Andrulko** [comment]
Hey @zeke, @alexandrtovmach! Excited to know you are going to launch another project soon! You have already existing translations, so I have submitted a configuration that will help to import them:
https://github.com/nodejs/nodejs.org/pull/2344

If you wish to start from scratch and ignore existing translations, feel free to remove `content_segmentation: 0` from configuration file. In this case, content will be segmented by sentences - this approach has also advantages, because this way there is much higher chance you'll get 100% Translation Memory matches in the future.

In case mentioned parameter is used, articles will be segmented by paragraphs (this approach allows us to import existing translations in the project).

**Comment by alexandrtovmach** [comment]
@zeke could you confirm that you given me permissions to create projects on behalf of _nodejs-crowdin_ account, [like here](https://support.crowdin.com/assigning-managers-to-several-projects/#advanced-management-feature-organization-plans). I can't create any new projects for nodejs now. It'll be strange to have a translation process of nodejs website under _alexandrtovmach_ account I think, so I need to create a new one. Or you can do that and add me in as a Manager.

@Andrulko Am I right?

I think good name for this project will be *nodejs-org-website*.

**Comment by alexandrtovmach** [comment]
In addition, I had a discussion with a nodejs.dev team, who works on a new version of website, and they would happy to go with one main translation process. But we found some doubts about migration process and content reusability in general.

Points:
- differences in the sites structure
- not too much content from the nodejs.org will be used and a lot of them will be modified or write from scratch

And now, this question is still open, but I think better to go with the separate Crowdin project like **nodejs-dev-website**. So @zeke could you also create one more project and add me as Manager or give me rights to create by my own.

**Comment by zeke** [comment]
Hey @alexandrtovmach. In order to create a new Crowdin project we'll need to first log in as @nodejs-crowdin. I have credentials for that, however my public key has changed since I last logged in, so I have opened a request to get my new public key added so I can log in again.

Apologies for all the roadblocks, and thanks for your persistence and patience. 👍

**Comment by alexandrtovmach** [comment]
@zeke Sure let's do this!)

**Comment by zeke** [comment]
Update: still waiting on someone from the build team to add my public key.

**Comment by alexandrtovmach** [comment]
@zeke Any updates?

**Comment by zeke** [comment]
I got stuck trying to decrypt the password for the @nodejs-crowdin user account, which owns the Crowdin account. @obensource do you happen to have access?

**Comment by alexandrtovmach** [comment]
@zeke @obensource any updates? That's blocker for me...

**Comment by alexandrtovmach** [comment]
😭

**Comment by obensource** [comment]
@zeke & @alexandrtovmach – sincere apologies for my extremely long delay here. Thank you for your passion for i18n! 🙏

I dug around and found the proper credentials for the Node.js account, @alexandrtovmach let me know how you'd like those delivered. @alexandrtovmach we could gpg via keybase if that works! 👍

**Comment by obensource** [comment]
@zeke I'll send them over to you right now.

**Comment by obensource** [comment]
Follow up: I should likely review our sharing policy – not sure if sharing GPG via keybase goes against it, but I'll get it to y'all however is appropriate. 🍻

**Comment by zeke** [comment]
Here we go! https://crowdin.com/project/nodejs-website

<img width="1061" alt="Screen Shot 2019-09-03 at 9 35 36 PM" src="https://user-images.githubusercontent.com/2289/64226085-cf4d2f80-ce92-11e9-938b-13abadab81ce.png">

@alexandrtovmach I've added you as a manger for this new project.

**Comment by zeke** [comment]
Thanks @obensource for the creds 👍

**Comment by alexandrtovmach** [comment]
@zeke @obensource Thank you.

About "sharing policy" ― I don't actually understand what is that. Could someone explain or give me some resources?

**Comment by alexandrtovmach** [comment]
@zeke I need to connect nodejs.org repo to CrowdIn here:
![Selection_311](https://user-images.githubusercontent.com/28801003/65161140-b2426000-da3f-11e9-9d3d-a7079e73cab0.jpg)

But I haven't enough rights in nodejs organiztion to be able set this integration

**Comment by zeke** [comment]
Hey @obensource I forget: what GitHub user is associated with the the `nodejs-crowdin` Crowdin user?

**Comment by alexandrtovmach** [comment]
@zeke Seems like it's called `nodejs`
![image](https://user-images.githubusercontent.com/28801003/65169694-b8d8d380-da4f-11e9-815a-0d8045f281f4.png)

**Comment by alexandrtovmach** [comment]
So we have _CrowdIn/Node.js_ project that's connected with _github/nodejs_
And we need to connect the same _github/nodejs_ with _CrowdIn/nodejs-website_

**Comment by alexandrtovmach** [comment]
@zeke If you have access to it, just go https://crowdin.com/project/nodejs-website/settings#integration and connect accounts. All other things I'll handle by my own

**Comment by alexandrtovmach** [comment]
@zeke any chance to help me with that?

**Comment by TiagoDanin** [comment]
> Hey @obensource I forget: what GitHub user is associated with the the `nodejs-crowdin` Crowdin user?

Github user is @nodejs-crowdin (https://github.com/nodejs/build/issues/1220#issuecomment-379916791)

**Comment by nschonni** [comment]
Any progress here, it would help with https://github.com/nodejs/nodejs.org/issues/2696

**Comment by alexandrtovmach** [comment]
@nschonni Unfortunately, I'm still blocked with access to github account

Wait for @zeke @obensource

**Comment by nschonni** [comment]
@zeke @obensource any progress? There have been more people trying to add additional languages and having a better way for them to start contributing would be great

**Comment by zeke** [comment]
I've added a new GitHub Personal Access Token to the https://crowdin.com/project/nodejs GitHub integration. Going to close this issue. Feel free to open more targeted issues if there's something here that still needs to be discussed.

**Comment by nschonni** [comment]
@zeke that isn't the correct project

**Comment by zeke** [comment]
@nschonni oh shoot thanks for pointing that out.

This morning I checked https://crowdin.com/project/nodejs-website/settings#integration and saw an existing GitHub integration there that had synced recently, so I assumed that was still working. But now I see that is integrating with the `alexandrtovmach / i18n` repo, not the canonical`nodejs/1i8n` repo.

I have now removed the existing `alexandrtovmach / i18n` integration, created a new PAT for GitHub user @nodejs-crowdin, and recreated the integration in https://crowdin.com/project/nodejs-website:

![Screen Shot 2020-03-13 at 2 28 25 PM](https://user-images.githubusercontent.com/2289/76660849-1cdbe080-6537-11ea-8062-30e70406f1ca.png)

cc @alexandrtovmach

**Comment by nschonni** [comment]
@zeke thanks 👍 
I think the `nodejs/i18n` isn't the right project either, should be `https://github.com/nodejs/nodejs.org`, but if the token in now in the `nodejs-website` Crowdin project, that's the important part

**Comment by alexandrtovmach** [comment]
@zeke Thank you
@nschonni About confusing with project names, that's my fault because firstly I had project for nodejs website translation (https://crowdin.com/project/nodejs-website), but than when the whole process was staled I changed it, to translate API docs. Going to fix everything soon
